Data/Vector/Generic/Mutable.hs

```diff
@@ -518,13 +518,28 @@ take :: MVector v a => Int -> v s a -> v s a
 {-# INLINE take #-}
 take n v = unsafeSlice 0 (min (max n 0) (length v)) v
 
+takeEnd :: MVector v a => Int -> v s a -> v s a
+{-# INLINE takeEnd #-}
+takeEnd n v = unsafeSlice i (min n' m) v
+  where
+    n' = max n 0
+    m = length v
+    i = max 0 (m - n)
```

Comment:
I guess it's safer to write `max 0 (m-n')`

Reply:
Agreed. Done